src/item.cpp

```diff
@@ -6406,3 +6412,18 @@ void item::set_birthday( const time_point bday )
 {
     this->bday = bday;
 }
+
+int check_freezing_temperature( const item &it )
+{
+    int temp = FREEZING_TEMPERATURE;
+    if( it.made_of( material_id( "alcohol" ) ) && it.made_of( material_id( "water" ) ) ) {
+        temp = 23; // -5C, wine, beer etc. (assumption: 10% ethanol) 
+    }
+    if( it.made_of( material_id( "alcohol" ) ) && !it.made_of( material_id( "water" ) ) ) {
+        temp = -22; // -30C, vodka, whiskey etc. (assumption: 40% ethanol)
+    }
+    if( it.type->comestible->freezing_point != temp ) {
```

Comment:
There's also a slight flaw: one can't make non-freezable alcohol items via json (due to this condition).

Reply:
You can set freezing point to absolute zero like in paper to do it.

Reply:
But not in case of `alcohol`. It's freezing temperature will always be overridden.

Reply:
Placement of this check and what is does is exact opposite of what you say. Its giving JSON value the privilege to become new freezing point if its different then what alcohol calcualtion returns.

Reply:
Aye, you're right. I somehow misread the condition (assumed early exit). Apologies.